### ISSUE TITLE
[BUG]: 

### ISSUE THREAD

**Issue by Beheshtiiman2** [pycaret/pycaret#3511]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

Hi, 
I wanna plot the confusion matrix and ROC for both training and test sets. But pycaret only generates the  respective plots for test set. I use version 3.0.0.rc3

**from pycaret.datasets import get_data
juice = get_data('juice')
from pycaret.classification import *
exp_name = setup(data = juice,  target = 'Purchase')
lr = create_model('lr')
plot_model(lr, plot = 'confusion_matrix', plot_kwargs = {'percent' : False}, use_train_data = False)
plot_model(lr, plot = 'confusion_matrix', plot_kwargs = {'percent' : False}, use_train_data = True)**

What is the problem here?
Thanks,

### Reproducible Example

```python
**from pycaret.datasets import get_data
juice = get_data('juice')
from pycaret.classification import *
exp_name = setup(data = juice,  target = 'Purchase')
lr = create_model('lr')
plot_model(lr, plot = 'confusion_matrix', plot_kwargs = {'percent' : False}, use_train_data = False)
plot_model(lr, plot = 'confusion_matrix', plot_kwargs = {'percent' : False}, use_train_data = True)**
```

### Expected Behavior

**from pycaret.datasets import get_data
juice = get_data('juice')
from pycaret.classification import *
exp_name = setup(data = juice,  target = 'Purchase')
lr = create_model('lr')
plot_model(lr, plot = 'confusion_matrix', plot_kwargs = {'percent' : False}, use_train_data = False)
plot_model(lr, plot = 'confusion_matrix', plot_kwargs = {'percent' : False}, use_train_data = True)**

### Actual Results

```python-traceback
**from pycaret.datasets import get_data
juice = get_data('juice')
from pycaret.classification import *
exp_name = setup(data = juice,  target = 'Purchase')
lr = create_model('lr')
plot_model(lr, plot = 'confusion_matrix', plot_kwargs = {'percent' : False}, use_train_data = False)
plot_model(lr, plot = 'confusion_matrix', plot_kwargs = {'percent' : False}, use_train_data = True)**
```

### Installed Versions

3.0.0.rc3

**Comment by rotemgb** [comment]
Hi There, I can have a look at it to see what's the issue

**Comment by ngupta23** [comment]
@rotemgb - Sure. That would be great.

**Comment by moezali1** [comment]
@Beheshtiiman2 It is not possible at the moment with `plot_model`. However if you want you can use the model to generate prediction on train set which is available through `get_config('X_train_transformed')`. You can retrieve the score from there and then plot confusion matrix manually.

I have tagged this issue as enhancement that we can plan in future as we get some space and time.

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
